src/utils/index.ts

```diff
@@ -150,3 +150,101 @@ export function shouldRenderEditButton(editButton: IConfigEditButton): boolean {
     // Priority list => 1. cslpEditButton query value 2.  Inside live preview  3. renderCslpButtonByDefault value selected by user
     return true;
 }
+
+export function getEditButtonPosition(
+    currentHoveredElement: HTMLElement | null,
+    defaultPosition: string | undefined
+): IEditButtonPosition {
+    if (!currentHoveredElement)
+        return { upperBoundOfTooltip: 0, leftBoundOfTooltip: 0 };
+
+    const cslpButtonPosition = currentHoveredElement.getAttribute(
+        "data-cslp-button-position"
+    );
+    if (cslpButtonPosition) {
+        return calculateEditButtonPosition(
+            currentHoveredElement,
+            cslpButtonPosition
+        );
+    }
+
+    // NOTE: position "top" and "top-left" will be the position of edit button if no default position passed in config
+    return calculateEditButtonPosition(
+        currentHoveredElement,
+        defaultPosition || "top"
+    );
+}
+
+function calculateEditButtonPosition(
+    currentHoveredElement: HTMLElement,
+    cslpButtonPosition: string
+): IEditButtonPosition {
+    const editButtonPosition: IEditButtonPosition = {
+        upperBoundOfTooltip: 0,
+        leftBoundOfTooltip: 0,
+    };
+    const currentRectOfElement = currentHoveredElement.getBoundingClientRect();
+    try {
+        switch (cslpButtonPosition) {
+            case "top-center":
+                editButtonPosition.upperBoundOfTooltip =
+                    currentRectOfElement.top - 40;
+                editButtonPosition.leftBoundOfTooltip =
+                    currentRectOfElement.width / 2 - 35.5;
```

Comment:
What are these values we had subtracted?

Reply:
Could we add these numbers to a variable?

Reply:
These numbers are related to the height and width of edit button. I don't think to to put these number into variable will be good idea but i can add a comment to better understand these numbers